### DISCUSSION TITLE
Optimal valve opening and energy savings

### DISCUSSION THREAD

**Discussion by tim4l** [jmcollin78/versatile_thermostat#711] [General]
Hello,
I'm using Versatile Thermostat in a flat heated by an individual gas boiler and hot water radiators.
I use SONOFF thermostatic valves.
On home assistant, I set up a ‘central’ Vtherm for the boiler and another VTherm (over_climate, with valve direct control and a cycle time set to 15 minutes) for each room to be heated.

I've noticed that the valves are never open more than 20% and I'm wondering whether this is optimal operation, particularly with in order to save energy (if the valves could be wide opened, radiators could release more heat, and the boiler could be switched on for less time and use less gas, I suppose).

Do you have any advice on this subject?

Anyway, thanks for this very useful integration!

**Comment by jmcollin78** [comment]
Hello @tim4l 

VTherm adjust the openness dynamically depending of the heat power needed. The right question is : is the target temperature reach ? Is it fast to reach the target ?

Yo ucan have advice here to tune your TPI algorithm: https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/algorithms.md#the-tpi-algorithm

You have also optimisations advices here: https://www.hacf.fr/optimisation-versatile-thermostat/ (in French).

If you need more advice, I will need regulation curve as explained here: https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/additions.md#regulation-curves-with-plotly-to-fine-tune-your-thermostat

**Reply by klaushipp** [reply]
Hi @jmcollin78 
I can't find an explanation how to create the plotly curves? 
Do you have a guide or something similar?

Thank you in advance! And of course for your integration.
I have a similar issue than others. I'm using the the sonoff TRVZB and have problems to find the right coefs. Still have overshoots of 1°C. To come a little bit closer to I think plotting the values would help. :)

**Reply by klaushipp** [reply]
Nevermind. Now I see, the graphs are just home assistant graphs. Stupid me. :)

**Comment by caiusseverus** [comment]
I have Sonoff TRVs and a gas boiler. I have 6 TRVZBs on the radiators downstairs controlled by 4 VTherms, and normal TRVs on the ones upstairs. I've been experimenting with different TPI settings as I found the suggested defaults to cause large overshoots.

I think this is mainly caused by the external coefficient combined with the thermal inertia of the radiators so I began by removing it completely by setting it to 0. I then started with the internal coefficient at 0.6 and waited until the following day to see how it behaved. I also have the minimum valve opening set to 5%, and the boiler will only fire up when at least 3 radiators want heat.

Here's an example: 

<img width="493" height="428" alt="image" src="https://github.com/user-attachments/assets/c04ef144-6e3b-4414-8743-35b922c4dc17" />

There is a certain amount of oscillation, so I'm gradually reducing the internal coefficient by small amounts - 0.05 or less each time to see if it will settle more evenly. One room oscillated around a point slightly lower than the set point, so I've added a tiny amount of external coefficient - only 0.001. I will increase this by 0.001 each time if it's not sufficient:

<img width="488" height="435" alt="image" src="https://github.com/user-attachments/assets/f7f5d072-61e5-4f38-ae81-fcad23351645" />

I don't know if it's possible to completely even these oscillations out with radiators due to the thermal mass of the system and the relatively non-controllable nature of the heat from the boiler - many boilers are set to a fixed flow temperature.

I'm also not convinced of the linearity of the response of radiators to the valve position. ie, 10% valve opening probably does not equal 10% power, but the exact response will depend on the flow temperature, and the position of the lock shield valve which also restricts the flow into and out of the radiator. The TRV can only ever allow as much flow as the lock shield does, so if that is quite restricted it may be that the radiator is at full power at 30 or 40% open or even less. 

My suspicion is that it might also be necessary to have a little hysteresis in the radiator control, so that the boiler does actually turn off. If you get the tpi set 'perfectly' for the radiators so that their output exactly matches the heat loss, they will always be calling for heat and the boiler will run constantly. Very few boilers are able to throttle down sufficiently to do this efficiently so will short cycle and waste gas. With a heat pump you probably can get it very close, because they are very good at providing low level heat over long periods.

This is very different to an electric fan heater or similar which stops providing heat as soon as it is switched off. 

As long as I can keep any oscillation to within a few tenths of a degree then I think that will be sufficient.
